sharding/proposer/service.go

```diff
@@ -27,31 +28,49 @@ type Proposer struct {
 	p2p          *p2p.Server
 	txpool       *txpool.TXPool
 	shardChainDb ethdb.Database
```

Comment:
we don't need `shardChainDb ethdb.Database` in Proposer struct, `shard *sharding.Shard` is the wrapper for it. Same for proposer and notary too